api/v3/Case.php

```diff
@@ -106,8 +106,11 @@ function civicrm_api3_case_create($params) {
     }
 
     // check for same contact id for edit Client
-    if (!empty($params['contact_id']) && !in_array($params['contact_id'], $origContactIds)) {
-      $mergedCaseId = CRM_Case_BAO_Case::mergeCases($params['contact_id'], $params['case_id'], $origContactId, NULL, TRUE);
+    if (!empty($params['contact_id'])) {
+      $mainContactId = CRM_Utils_Array::first($params['contact_id']);
```

Comment:
This doesn't look right. You seem to be assuming the value is an array here, but it might not be.